param/parameterized.py

```diff
@@ -4383,51 +5081,125 @@ def __setstate__(self, state):
 
 class Parameterized(metaclass=ParameterizedMetaclass):
     """
```

Comment:
The Numpy docstring guide seems to say that you should document `__init__` in the class docstring (here), see https://numpydoc.readthedocs.io/en/latest/format.html#class-docstring. That's what Pandas does for example https://github.com/pandas-dev/pandas/blob/b8a4691647a8850d681409c5dd35a12726cd94a1/pandas/core/frame.py#L505.
So the `Parameters` section. you added to `__init__` should be moved to the class docstring. Apparently you could also add an `Attributes` section to document `name` and `param`.

Reply:
Thx. I'm a little bit in doubt what to do here because 

- the numpy recommendation was created long time before for example VS Code was a thing. In VS Code I believe I have seen the class docstring and the `__init__` docstring being used at different times by pylance tooltips.
- In param docs the `__init__` method is documented. See for example https://param.holoviz.org/reference/param/generated/param.Action.html#param.Action.__init__. If I remove the docstring from `__init__` to param docs will be empty there. Might not be a problem.

What should I do?

Reply:
Someone will need to look more into that, without forgetting Jupyter as it's a major target for HoloViz packages.

> In param docs the __init__ method is documented

Yes I already mentioned that documenting the Parameters is tricky.

Reply:
I tried it in VSCode and JupyterLab and was satisfied with the experience, the longer class docstring showing up on `Parameterized` and the shorter `__init__` one when instantiating a Parameterized object.